js/plugins/iq.js

```diff
@@ -192,145 +203,111 @@ var IqPlugin= function (framework)
+        this.ooyalaReporter.reportPlayRequested(autoPlay);
         break;
       //OO.EVENTS.PLAYHEAD_TIME_CHANGED -> OO.Analytics.EVENTS.VIDEO_STREAM_POSITION_CHANGED.
       case OO.Analytics.EVENTS.VIDEO_STREAM_POSITION_CHANGED:
-        if (params && params[0])
-        {
+        if (params && params[0] && params[0].streamPosition > 0)
+        {            
           currentPlayheadPosition = params[0].streamPosition;
-          if (currentPlayheadPosition > 0)
+          if (playingInstreamAd)
           {
-            if (this.ooyalaReporter)
-            {
-              /* TODO: disable for now as this is already reported by reporter.js in core */
-              //var currentPlayheadPositionMilli = currentPlayheadPosition * 1000;
-              //this.ooyalaReporter.reportPlayHeadUpdate(currentPlayheadPositionMilli);
-              //OO.log("IQ: Reported: reportPlayHeadUpdate() with args: " + Math.floor(currentPlayheadPosition * 1000));
-            }
-            else
+            var totalTime = params[0].totalStreamDuration;
+            var percentPlayed = 0;
+            var reportQuartile = false;
+            if (totalTime != null && totalTime > 0 )
             {
-              OO.log("IQ: Tried reporting event: " + OO.Analytics.EVENTS.VIDEO_STREAM_POSITION_CHANGED +
-                     " but ooyalaReporter is: " + this.ooyalaReporter);
+              if (currentPlayheadPosition >= 0.25 * totalTime && !adFirstQuartile){
+                percentPlayed = 0.25;
+                adFirstQuartile = true;
+                reportQuartile = true;
+              } 
+              else if (currentPlayheadPosition >= 0.50 * totalTime && !adSecondQuartile)
+              {
+                percentPlayed = 0.50;
+                adSecondQuartile = true;
+                reportQuartile = true;
+              } 
+              else if (currentPlayheadPosition >= 0.75 * totalTime && !adThirdQuartile)
+              {
+                percentPlayed = 0.75;
+                adThirdQuartile = true;
+                reportQuartile = true;
+              } 
+              else if (currentPlayheadPosition >= 1.0 * totalTime && !adLastQuartile)
+              {
+                percentPlayed = 1.00;
+                adLastQuartile = true;
+                reportQuartile = true;
+              }
+
+              if (reportQuartile){
+                OO.log("IQ: Reported: reportCustomEvent() for event: adPlaythrough with args:" + JSON.stringify(percentPlayed));
+                this.ooyalaReporter.reportCustomEvent(eventName, {"adEventName": "adPlaythrough", "percent": percentPlayed });
+              }
             }
+          } 
+          else 
+          {
+            var currentPlayheadPositionMilli = currentPlayheadPosition * 1000;
+            this.ooyalaReporter.reportPlayHeadUpdate(currentPlayheadPositionMilli);
+            OO.log("IQ: Reported: reportPlayHeadUpdate() with args: " + Math.floor(currentPlayheadPosition * 1000));
```

Comment:
Should we log the calculated value (currentPlayheadPositionMilli) instead of recalculating for the log? We can floor currentPlayheadPositionMilli if necessary.

Reply:
@aeng7 yes, logging currentPlayheadPositionMilli directly now